app.py

```diff
@@ -184,28 +176,102 @@ def main():
     pull_cache_from_hub = arguments.pull_cache_from_hub
 
     # Initialize the interface and grab the UI-provided arguments
-    dataset_args = display_initial_UI()
-
-    # TODO: Make this less of a weird outlier.
-    show_perplexities = st.sidebar.checkbox("Show text perplexities")
-
-    # Initialize the main DMT class with the UI-provided arguments
-    # When using the app (this file), try to use cache by default.
-    dstats = dmt_cls(**dataset_args, use_cache=True)
-    display_title(dstats)
-    # Get the widget functionality for the different measurements.
-    load_prepare_list, display_list = get_widgets(dstats)
-    # Load/Prepare the DMT widgets.
-    loaded_dstats = load_or_prepare_widgets(dstats, load_prepare_list,
-                                            show_perplexities, live=live,
-                                            pull_cache_from_hub=pull_cache_from_hub)
-    # After the load_or_prepare functions are run,
-    # we should have a cache for each measurement widget --
-    # either because it was there already,
-    # or we computed them (which we do when not live).
-    # Write out on the UI what we have.
-    display_measurements(dataset_args, display_list, loaded_dstats,
-                         show_perplexities)
+    with gr.Blocks() as demo:
+        widget_list = [widgets.DatasetDescription(DATASET_NAME_TO_DICT),
+                       widgets.GeneralStats(),
+                       widgets.LabelDistribution(),
+                       widgets.TextLengths(),
```

Comment:
After this is called, I get the following error and it crashes:



2022-11-29 15:13:41.234 python[43481:9115201] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'
*** First throw call stack:
(
        0   CoreFoundation                      0x00007fff3460cdd7 __exceptionPreprocess + 250
        1   libobjc.A.dylib                     0x00007fff6d06d5bf objc_exception_throw + 48
        2   CoreFoundation                      0x00007fff346355bc -[NSException raise] + 9
        3   AppKit                              0x00007fff3183296c -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 310
        4   AppKit                              0x00007fff3181a3d2 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1416
        5   AppKit                              0x00007fff31819e43 -[NSWindow initWithContentRect:styleMask:backing:defer:] + 42
        6   _macosx.cpython-39-darwin.so        0x0000000186f845ac -[Window initWithContentRect:styleMask:backing:defer:withManager:] + 76
        7   _macosx.cpython-39-darwin.so        0x0000000186f87a49 FigureManager_init + 281
        8   python                              0x0000000106c4bbac wrap_init + 12
        9   python                              0x0000000106bcf8ef wrapperdescr_call + 911
        10  python                              0x0000000106cdf147 _PyEval_EvalFrameDefault + 52951
        11  python                              0x0000000106bc28b4 _PyFunction_Vectorcall + 420
        12  python                              0x0000000106c4b50a slot_tp_init + 346
        13  python                              0x0000000106c50ce0 type_call + 272
        14  python                              0x0000000106cdf147 _PyEval_EvalFrameDefault + 52951
        15  python                              0x0000000106bc28b4 _PyFunction_Vectorcall + 420
        16  python                              0x0000000106bc79ae method_vectorcall + 158
        17  python                              0x0000000106cdbd9e _PyEval_EvalFrameDefault + 39726
        18  python                              0x0000000106bc2be2 _PyFunction_Vectorcall + 1234
        19  python                              0x0000000106bc79ae method_vectorcall + 158
        20  python                              0x0000000106ce062c _PyEval_EvalFrameDefault + 58300
        21  python                              0x0000000106bc2be2 _PyFunction_Vectorcall + 1234
        22  python                              0x0000000106ce062c _PyEval_EvalFrameDefault + 58300
        23  python                              0x0000000106bc2be2 _PyFunction_Vectorcall + 1234
        24  python                              0x0000000106ce062c _PyEval_EvalFrameDefault + 58300
        25  python                              0x0000000106bc2be2 _PyFunction_Vectorcall + 1234
        26  python                              0x0000000106cd9098 _PyEval_EvalFrameDefault + 28200
        27  python                              0x0000000106bc28b4 _PyFunction_Vectorcall + 420
        28  python                              0x0000000106cd913e _PyEval_EvalFrameDefault + 28366
        29  python                              0x0000000106bc28b4 _PyFunction_Vectorcall + 420
        30  python                              0x0000000106cda8dd _PyEval_EvalFrameDefault + 34413
        31  python                              0x0000000106bc2be2 _PyFunction_Vectorcall + 1234
        32  python                              0x0000000106bc79ae method_vectorcall + 158
        33  python                              0x0000000106cd9098 _PyEval_EvalFrameDefault + 28200
        34  python                              0x0000000106bc2be2 _PyFunction_Vectorcall + 1234
        35  python                              0x0000000106cd9098 _PyEval_EvalFrameDefault + 28200
        36  python                              0x0000000106bc2be2 _PyFunction_Vectorcall + 1234
        37  python                              0x0000000106d17589 _PyObject_VectorcallTstate.4590 + 73
        38  python                              0x0000000106d17471 context_run + 81
        39  python                              0x0000000106c204d1 cfunction_vectorcall_FASTCALL_KEYWORDS + 97
        40  python                              0x0000000106cddb21 _PyEval_EvalFrameDefault + 47281
        41  python                              0x0000000106bc28b4 _PyFunction_Vectorcall + 420
        42  python                              0x0000000106cda8dd _PyEval_EvalFrameDefault + 34413
        43  python                              0x0000000106bc28b4 _PyFunction_Vectorcall + 420
        44  python                              0x0000000106cda8dd _PyEval_EvalFrameDefault + 34413
        45  python                              0x0000000106bc28b4 _PyFunction_Vectorcall + 420
        46  python                              0x0000000106bc7aef method_vectorcall + 479
        47  python                              0x0000000106dc222a t_bootstrap + 186
        48  python                              0x0000000106d5f804 pythread_wrapper + 36
        49  libsystem_pthread.dylib             0x00007fff6e41a109 _pthread_start + 148
        50  libsystem_pthread.dylib             0x00007fff6e415b8b thread_start + 15
)
libc++abi.dylib: terminating with uncaught exception of type NSException
zsh: abort      python app.py

Reply:
I got that during development because of the use of seaborn (not thread safe). I do not get that locally. Could it be because I recached all of the results?